meshnets/utils/datasets.py

```diff
@@ -1,90 +1,197 @@
-    files retrieved from disk are torch_geometric Data.
+            processed_file_path = os.path.join(self.get_sample_path(i),
+                                               self.graph_file_name)
 
-    It also offer a new property called 'num_label_features' following
-    the implementation of 'num_node_features' and 'num_edge_features'
-    in torch_geometric Dataset.
-    """
+            torch.save(processed_graph, processed_file_path)
 
-    def __init__(self, data_dir, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.from_disk_dataset = FromDiskDataset(data_dir=data_dir)
+    def remove_files(self, file_name_to_delete: str) -> None:
+        """Remove the required file in each sample directory."""
+
+        for i in range(self.len()):
+            file_path_to_delete = os.path.join(self.get_sample_path(i),
+                                               file_name_to_delete)
+            os.remove(file_path_to_delete)
 
     def len(self) -> int:
-        """Returns the number of graphs stored in the dataset.
-        
-        This is the number of files in the dataset from disk."""
-        return len(self.from_disk_dataset)
+        """Return the number of samples in the dataset."""
+        return len(self.samples)
+
+    def get_sample_path(self, idx: int) -> str:
+        """Get the path to the sample directory at the given index."""
+        return os.path.join(self.root_dir, self.samples[idx])
+
+    def get_mesh_path(self, idx: int) -> str:
+        """Get the path to the mesh file at the given index."""
+
+        sample_path = self.get_sample_path(idx)
+        mesh_path = os.path.join(sample_path, self.mesh_file_name)
+
+        if not os.path.isfile(mesh_path):
+            #raise FileNotFoundError(mesh_path)
+            warnings.warn(f"File '{mesh_path}' does not exist.")
+
+        return mesh_path
+
+    def get_graph_path(self, idx: int) -> str:
+        """Get the path to the graph file at the given index."""
+
+        sample_path = self.get_sample_path(idx)
+        graph_path = os.path.join(sample_path, self.graph_file_name)
+
+        if not os.path.isfile(graph_path):
+            #raise FileNotFoundError(graph_path)
+            warnings.warn(f"File '{graph_path}' does not exist.")
+
+        return graph_path
 
     def get(self, idx: int) -> Data:
-        """Gets the data object at index :obj:`idx`.
+        """Get an element from the dataset at the given index.
         
-        This is the data object at index :obj:`idx` in the dataset from disk."""
-        return self.from_disk_dataset[idx]
+        The only things that exist in RAM at this point are the paths 
+        to the samples. This method loads and returns the sample from disk."""
+
+        graph_path = self.get_graph_path(idx)
+        return torch.load(graph_path)
+
+    def get_stats(self):
+        """Compute and return the dataset statistics."""
+
+        sample = self[0]
```

Comment:
This is very cryptic.

Reply:
This extracts the first dataset sample to obtain the data shapes